src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,79 @@
 package mate.academy.lib;
 
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Injector {
     private static final Injector injector = new Injector();
 
     public static Injector getInjector() {
         return injector;
     }
 
+    Map<Class<?>, Object> instances = new HashMap<>();
```

Comment:
Make Interface Implementations map a class field. You can fill it in using `Map.of()`. Currently, the map is being created inside the `findImplementation` method, which is not efficient as it gets created on each call.